core/node/node_framework/src/service/tests.rs

```diff
@@ -0,0 +1,235 @@
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use futures::lock::Mutex;
+use tokio::runtime::Runtime;
+
+use crate::service::{
+    ServiceContext, StopReceiver, Task, WiringError, WiringLayer, ZkStackService,
+};
+
+// ZkStack Service's `new()` method has to have a check for nested runtime.
+#[test]
+fn test_new_with_nested_runtime() {
+    let runtime = Runtime::new().unwrap();
+
+    runtime.block_on(async {
+        assert_eq!(
+            ZkStackService::new().unwrap_err().to_string(),
+            "Detected a Tokio Runtime. ZkStackService manages its own runtime and does not support nested runtimes".to_string()
+        );
+    })
```

Comment:
Btw here is an example where you can detach logic from assertion.

You can have `let run_result = runtime.block_on(...);` and then check if `run_result` is an expected error.

Reply:
why this is better?

Reply:
https://github.com/matter-labs/zksync-era/pull/1102#discussion_r1498776007